### PR TITLE
Add plotly-esque point data info to pie & sankey event data

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/1875

### CODE REVIEW

File: src/traces/sankey/render.js [hunk 1]
```diff
@@ -150,13 +150,17 @@ function sankeyModel(layout, d, i) {
...
     var tc = tinycolor(l.color);
     var basicKey = l.source.label + '|' + l.target.label;
     var foundKey = uniqueKeys[basicKey];
     uniqueKeys[basicKey] = (foundKey === void(0) ? foundKey : 0) + 1;
     var key = basicKey + (foundKey === void(0) ? '' : '__' + foundKey);
 
+    // for event data
+    l.trace = d.trace;
+    l.curveNumber = d.trace.index;
+    l.pointNumber = 'TODO';
```
Comment (etpinard) [inline src/traces/sankey/render.js]: @monfera is there a _easy_ way to grab the `pointNumber` value off the sankey link & node d3-esque data?

File: test/jasmine/tests/sankey_test.js [hunk 2]
```diff
@@ -410,4 +410,101 @@ describe('sankey tests', function() {
             .then(done);
         });
     });
+
+    describe('Test hover/click event data:', function() {
```
Comment (etpinard) [inline test/jasmine/tests/sankey_test.js]: Sankey event data wasn't tested anywhere before :arrow_heading_up: :upside_down_face:

File: src/traces/sankey/render.js (hunk 1 above)
Comment (etpinard) [inline src/traces/sankey/render.js]: done in 15dc324

**Comment by etpinard** [comment]
@alexcjohnson might as well include this in `1.29.0` too.

File: test/jasmine/tests/pie_test.js [hunk 3]
```diff
@@ -392,18 +394,22 @@ describe('Test event property of interactions on a pie plot:', function() {
             expect(typeof trace).toEqual(typeof {}, 'points.trace');
 
             var pt = futureData.points[0];
-            expect(Object.keys(pt)).toEqual([
+            expect(Object.keys(pt)).toEqual(jasmine.arrayContaining([
```
Comment (alexcjohnson) [inline test/jasmine/tests/pie_test.js]: nice, didn't know about `arrayContaining`!

**Comment by alexcjohnson** [comment]
No comments from me, very nice! 💃

File: src/traces/sankey/render.js (hunk 1 above)
Comment (monfera) [inline src/traces/sankey/render.js]: Thanks @etpinard I missed the question on the road, now see it's done, very nice! 🙇

**Pull request merged**